### PR TITLE
chore(deps): upgrade o-f/api for new types

### PR DESCRIPTION
**Pull request by estroz**
**Description of the change:** upgrade `operator-framework/api`

**Motivation for the change:** https://github.com/operator-framework/api/pull/198, https://github.com/operator-framework/api/pull/202, https://github.com/operator-framework/api/pull/203

**Reviewer Checklist**
- [ ] Implementation matches the proposed design, or proposal is updated to match implementation
- [ ] Sufficient unit test coverage
- [ ] Sufficient end-to-end test coverage
- [ ] Docs updated or added to `/doc`
- [ ] Commit messages sensible and descriptive

/cc @dinhxuanvu

### CODE REVIEW

**Review by dinhxuanvu — APPROVED** [review]
/lgtm

File: go.mod [hunk 1]
```diff
@@ -64,5 +64,5 @@ replace (
 	github.com/openshift/client-go => github.com/openshift/client-go v0.0.0-20200326155132-2a6cd50aedd0 // release-4.5
 	// Patch for a race condition involving metadata-only
 	// informers until it can be resolved upstream:
-	sigs.k8s.io/controller-runtime v0.9.2 => github.com/benluddy/controller-runtime v0.9.3-0.20210720171926-9bcb99bd9bd3
+	sigs.k8s.io/controller-runtime v0.10.0 => github.com/benluddy/controller-runtime v0.9.3-0.20210720171926-9bcb99bd9bd3
```
Comment (timflannagan) [inline go.mod:67]: We'll need to drop this commit and rebase against master now that #2368 landed.

Comment (estroz) [reply go.mod:67]: Yep, just waiting on https://github.com/operator-framework/api/pull/203 to merge then I'll update the pin here and rebase onto master.

File: go.mod [hunk 2]
```diff
@@ -13,32 +13,32 @@ require (
...
 	github.com/mikefarah/yq/v3 v3.0.0-20201202084205-8846255d1c37
 	github.com/mitchellh/hashstructure v1.0.0
-	github.com/mitchellh/mapstructure v1.1.2
+	github.com/mitchellh/mapstructure v1.4.1
 	github.com/onsi/ginkgo v1.16.4
 	github.com/onsi/gomega v1.15.0
 	github.com/openshift/api v0.0.0-20200331152225-585af27e34fd
 	github.com/openshift/client-go v0.0.0-20200326155132-2a6cd50aedd0
-	github.com/operator-framework/api v0.10.3
+	github.com/operator-framework/api v0.10.8-0.20211210205029-40cb9fd4036a
```
Comment (timflannagan) [inline go.mod:28]: +1 - note: this maps to https://github.com/operator-framework/api/commit/40cb9fd4036a.

**Pull request merged**